### ISSUE TITLE
Add eval-quoted

### ISSUE THREAD

**Issue by NightMachinery** [AdrieanKhisbe/diractions#20]
Currently,

```
somedirac , count-arg ‘1 2 3’
# outputs 3
```

I propose adding:

```
somedirac ,, count-arg ‘1 2 3’
# outputs 1
somedirac ,, echo ‘*’
# outputs *
```

You can use these functions for eval-quoting:

```
reval () {
 eval "$(gquote "$@")"
}
gquote () {
 print -r -- "${(q+@)@}"
}
```

**Comment by AdrieanKhisbe** [comment]
Hello again @NightMachinary, 

Just released [v0.17.0](https://github.com/AdrieanKhisbe/diractions/releases/tag/v0.17.0)
Tell me if the `quoted-exec` suits your needs. :)

(I followed your implementation suggestion.  Be aware it's only working starting from zsh5.3)

**Comment by NightMachinery** [comment]
@AdrieanKhisbe Thanks for all the work! 💙
